mailpoet/lib/Automation/Integrations/MailPoet/Analytics/Controller/AutomationEmailController.php

```diff
@@ -0,0 +1,92 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\MailPoet\Analytics\Controller;
+
+use MailPoet\Automation\Engine\Data\Automation;
+use MailPoet\Automation\Engine\Storage\AutomationStorage;
+use MailPoet\Automation\Integrations\MailPoet\Actions\SendEmailAction;
+use MailPoet\Entities\NewsletterEntity;
+use MailPoet\Newsletter\NewslettersRepository;
+
+class AutomationEmailController {
+
+
+  /** @var AutomationStorage */
+  private $automationStorage;
+
+  /** @var NewslettersRepository */
+  private $newslettersRepository;
+
+  public function __construct(
+    AutomationStorage $automationStorage,
+    NewslettersRepository $newslettersRepository
+  ) {
+    $this->automationStorage = $automationStorage;
+    $this->newslettersRepository = $newslettersRepository;
+  }
+
+  /**
+   * @param Automation $automation
+   * @param \DateTimeImmutable $after
+   * @param \DateTimeImmutable $before
+   * @return NewsletterEntity[]
+   */
+  public function getAutomationEmailsInTimeSpan(Automation $automation, \DateTimeImmutable $after, \DateTimeImmutable $before): array {
+    $automationVersions = $this->automationStorage->getAutomationVersionDates($automation->getId());
+    usort(
+      $automationVersions,
+      function (array $a, array $b) {
+        return $a['created_at'] <=> $b['created_at'];
+      }
+    );
+
+    // filter automations that were created before the after date
+    $versionIds = [];
+    foreach ($automationVersions as $automationVersion) {
+      if ($automationVersion['created_at'] > $before) {
+        break;
+      }
+      if (!$versionIds || $automationVersion['created_at'] < $after) {
```

Comment:
Could you please explain why do you add a version to the result if it doesn't match the `$after` constraint?

Reply:
But the `$automationVersion['created_at'] < $after` condition would also apply to versions 1 and 2 in your example. So it will initialize the `$versionIds` array several times until the correct version is put there. It works, but is a bit obfuscated.

Reply:
Hmm... if the `$automationVersions` array would be ordered reverse, maybe this looks better?
```
foreach ($automationVersions as $automationVersion) {
      if ($automationVersion['created_at'] > $before) {
        continue;
      }
      $versionIds[] = (int)$automationVersion['id'];
      if ($automationVersion['created_at'] < $after) {
      	break;
      }
}
```
What do you think @alex-mailpoet

Reply:
discussed via slack and fixed in 3fc69d05e8608f7eb83ed571fe5d45612bb18b7f